### PR TITLE
feat: Add API Routes and added axios version 1.6.7

### PR DESCRIPTION
**Pull request by sidarth-23**
## Summary
- Add types for login success and error responses. 
- Added BASE_URL to use environment variable or default to 'http://127.0.0.1:8000/api/'

- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [ ] Attached Screenshots / Videos 
- [x] All Relevant Documents added

### CODE REVIEW

File: apps/admin/api-routes/base-url.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const BASE_URL = process.env.BASE_URL || 'http://127.0.0.1:8000/api/';
```
Comment (yaju09) [inline apps/admin/api-routes/base-url.ts]:
```suggestion
export const BASE_URL = process.env.BASE_URL || 'http://127.0.0.1:8000/api/';
```
will it be better to have diff base url's for diff env's ?

BASE_URL_LOCAL
BASE_URL_STAGING
BASE_URL_PROD

**Review by yaju09 — CHANGES_REQUESTED** [review]

File: apps/admin/api-routes/base-url.ts (hunk 1 above)
Comment (yogai53) [reply apps/admin/api-routes/base-url.ts]: @yaju09 we don't have any plan for infrastructure. So let's not plan for environment related works.

Comment (sidarth-23) [reply apps/admin/api-routes/base-url.ts]: @yogai53 Then should I hard code the url for now ?

**Review by yogai53 — DISMISSED** [review]

File: apps/admin/api-routes/base-url.ts (hunk 1 above)
Comment (yaju09) [reply apps/admin/api-routes/base-url.ts]: @yogai53 got it Yoga

Comment (yaju09) [reply apps/admin/api-routes/base-url.ts]: @sidarth-23 the above code is fine in that case.

**Review by yaju09 — APPROVED** [review]

**Review by yogai53 — APPROVED** [review]

**Pull request merged**